gci/oci.py

```diff
@@ -106,34 +106,3 @@ def component_descriptor_from_tarfileobj(
           raise ValueError('Component Descriptor appears to be empty')
 
         return ocm.ComponentDescriptor.from_dict(raw_dict)
-
-
-def image_ref_with_digest(
```

Comment:
Actually, this function is being used [here](https://github.com/open-component-model/delivery-service/blob/cf8ea385c6b6940c0bc0a2ad95313228c8f5d515/bdba/util.py#L299) in delivery-service. That being said, I'm not against dropping this function (here), but we'll have to adjust delivery-service accordingly first.

Reply:
indeed. I think moving this function to delivery-service will make sense. maybe we can drop some codepaths